### ISSUE TITLE
expected type *v1beta1.Deployment, but watch event object had type *v1.Deployment

### ISSUE THREAD

**Issue by gjcarneiro** [kubernetes/kube-state-metrics#719]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:

Deployed `quay.io/coreos/kube-state-metrics:v1.6.0-rc.0`

I keep getting these messages in logs, over and over:

```
E0410 15:22:20.836131       1 reflector.go:329] k8s.io/kube-state-metrics/internal/collector/builder.go:482: expected type *v1beta1.Deployment, but watch event object had type *v1.Deployment
E0410 15:22:20.842987       1 reflector.go:329] k8s.io/kube-state-metrics/internal/collector/builder.go:482: expected type *v1beta1.Deployment, but watch event object had type *v1.Deployment
E0410 15:22:20.876726       1 reflector.go:329] k8s.io/kube-state-metrics/internal/collector/builder.go:482: expected type *v1beta1.Deployment, but watch event object had type *v1.Deployment
```

**What you expected to happen**:

It shouldn't log errors.

**How to reproduce it (as minimally and precisely as possible)**:

This my deployment yaml file:

[kube-state-metrics.yaml.txt](https://github.com/kubernetes/kube-state-metrics/files/3064561/kube-state-metrics.yaml.txt)

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.13.3
- Kube-state-metrics image version v1.6.0-rc.0

**Comment by tariq1890** [comment]
Thanks for reporting this issue. We used `v1beta1` apiGroup for the sake of supporting older versions of k8s. 

If we are to move the apiGroup to core `v1` , we would need to look into updating the compatibility of matrix of kube-state-metrics v1.6.0.

Thoughts @brancz?

**Comment by brancz** [comment]
Is that so? Workloads went v1 in 1.9 if I recall correctly, so our compatibility matrix is correct as is (it says we don't even make any support claims about 1.9 at this point). Is the `extensions` API still available in the latest versions?

**Comment by tariq1890** [comment]
So I checked `release-1.14`. The `extensions` API is still there but  it does say that `extensions/v1beta1` is deprecated in favour of `apps/v1beta2/` https://github.com/kubernetes/api/blob/d687e77c8ae999eab9ebad49439d16e61add0df0/extensions/v1beta1/types.go#L82

It doesn't seem to explain this error though. Will have to dig deeper.

**Comment by tariq1890** [comment]
Please disregard my earlier comments.

This should be fixed by #720

**Comment by tariq1890** [comment]
I think this can be closed now.
